### ISSUE TITLE
The fill method of PlasticView is ambiguously named

### ISSUE THREAD

**Issue by frankdilo** [BendingSpoons/katana-swift#77]
It has happened to me a few times that I wanted to use the [`fill`](https://github.com/BendingSpoons/katana-swift/blob/master/Katana/Plastic/PlasticView+Convenience.swift#L62) method and got surprised by the behaviour. I called it this way `view.fill(top: anchor1, left: anchor2, bottom: anchor3, right: anchor4)` and expected my view to stretch to the anchors, instead I got a square view.

This is because `fill` stretches the view preserving the aspect ratio (1, by default). I think the method should be renamed to something like `fillPreservingAspectRatio` to avoid confusion, or at least the `scale` argument shouldn't be optional.

Moreover, I think it would be useful to have a `fill` method that does **not** preserve the aspect ratio.

**Comment by bolismauro** [comment]
I guess the problem is the misleading name.
That method is meant to be used to make the view as big as possible given the constraint. Probably `fill` is the wrong name. 

I'd change the name of this method (any idea about the name? It should mean `use as much space as possible between these anchors, using this aspect ratio`)

And we can add another method which is `fill(top:left:bottom:right:insets:)` that does what the current `fill` does

**Comment by frankdilo** [comment]
Given your definition, I think a better method name would be `scale(top:left:bottom:right:aspectRatio)`, or `scaleBetween(top:left:bottom:right:aspectRatio)`. 

I agree on adding the `fill(top:left:bottom:right:insets:)` method.

**Comment by bolismauro** [comment]
I don't really like `scale`. In Plastic it means something completely different

**Comment by madbat** [comment]
We could mimic `UIView`'s `contentMode` (the property that tells a view if/how its content should be scaled/stretched to match the view's frame).

So we could have a single `fill(top:left:bottom:right:mode:)` method, where `mode` could be something like:
```swift
enum FillMode {
  case stretch // meaning: do not preserve aspect ratio
  case fit // meaning: preserve aspect ratio and do not cut off parts of the view (like our current fill method)
  case fill // meaning: preserve aspect ratio and cut off parts of the view in order not to have blank space left
}
```

Does it make sense? 🤔

**Comment by sroddy** [comment]
@madbat what would be the aspect ratio in this case?

**Comment by madbat** [comment]
@sroddy ha! Good point.

I'd go with
```swift
enum FillMode {
  case stretch // meaning: do not preserve aspect ratio
  case fit(preservingAspectRatio: CGFloat) // meaning: preserve aspect ratio and do not cut off parts of the view (like our current fill method)
  case fill(preservingAspectRatio: CGFloat) // meaning: preserve aspect ratio and cut off parts of the view in order not to have blank space left
}
```

**Comment by bolismauro** [comment]
Seems good to me. I don't see use cases for fill but I guess it make sense to add them.
I still don't like the name
`fill(...mode: stretch)`. Is weird to me :)

**Comment by frankdilo** [comment]
I like @madbat's proposal, and think `stretch` should be default `FillMode`.

**Comment by bolismauro** [comment]
Discussing with @frankdilo and @sroddy we realized that the best thing to do it two 
- remove the current fill method with aspect ratio
- add a new method `fill(top:left:bottom:right:)`
- add a new method `fit(top:left:bottom:right:aspectRatio)`

**Comment by bolismauro** [comment]
Fixed by #83
